src/gui/userstatus.cpp

```diff
@@ -30,16 +30,37 @@ Q_LOGGING_CATEGORY(lcUserStatus, "nextcloud.gui.userstatus", QtInfoMsg)
 
 UserStatus::UserStatus(QObject *parent)
     : QObject(parent)
-    , _message("")
 {
 }
 
 UserStatus::Status UserStatus::stringToEnum(const QString &status) const 
 {
+    // it needs to match the Status enum
+    const QHash<QString, Status> preDefinedStatus{{"online", Online},
```

Comment:
@camilasan And, I think, this could just as well be implemented via `if`/`else if` for strings:
`if (statusString == "dnd") { return DoNotDisturb; }` and so on... Or, even, `std::find` on array of enum strings as proposed in enumToUserString. The search is linear in that case, but, the data size is so small, so, I can hardly see any significant benefit of using `QHash` here.

Reply:
I don't think it's an algorithmic performance thing, it's mainly a question of style I'd say. It's a table look up so it makes sense to have a table to look into. :-)

IME there's less cognitive load in reading the table look up version months from now than the if / else if version + factor in the lower case + factor in the missing status.